src/eval.c

```diff
@@ -302,11 +328,64 @@ sds luaCreateFunction(client *c, robj *body) {
         return dictGetKey(de);
     }
 
+    /* Handle shebang header in script code */
+    ssize_t shebang_len = 0;
+    if (!strncmp(body->ptr, "#!", 2)) {
```

Comment:
Maybe extract this to a function?

Reply:
This logic is part of our "create function" logic which does input validation etc. I agree that once we have multiple engines it'll make sense to break this up, but then I'm not sure exactly how. So for now I think it's good to keep this tightly bound to `luaCreateFunction()`.